classes/ObjectModel.php

```diff
@@ -1990,4 +1993,29 @@ public static function disableCache()
     {
         ObjectModel::$cache_objects = false;
     }
+
+    /**
+     * Return HtmlFields for object
+     */
+    public function getHtmlFields()
+    {
+        if (!empty($this->def) && is_array($this->def) && array_key_exists('table', $this->def)) {
```

Comment:
In order to avoid too much indentation, maybe you should return directly if this test is true ?